bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyHttpApi.java

```diff
@@ -379,154 +348,210 @@ private void setRelayEvents() throws IOException {
-                            SHELLY_API_EVENTURL_BTN2_OFF));
-                } else {
-                    // Standard relays: btn_xxx URLs
-                    logger.debug("Set Relay event urls");
-                    request(buildSetEventUrl(lip, localPort, deviceName, index, EVENT_TYPE_RELAY,
-                            SHELLY_API_EVENTURL_BTN_ON));
-                    request(buildSetEventUrl(lip, localPort, deviceName, index, EVENT_TYPE_RELAY,
-                            SHELLY_API_EVENTURL_BTN_OFF));
+            setEventUrl(EVENT_TYPE_ROLLER, 0, config.eventsRoller, SHELLY_EVENT_ROLLER_OPEN, SHELLY_EVENT_ROLLER_CLOSE,
+                    SHELLY_EVENT_ROLLER_STOP);
+        } else if (profile.isDimmer) {
+            // 2 set of URLs
+            setEventUrl(EVENT_TYPE_LIGHT, index, config.eventsButton, SHELLY_EVENT_BTN1_ON, SHELLY_EVENT_BTN1_OFF,
+                    SHELLY_EVENT_BTN2_ON, SHELLY_EVENT_BTN2_OFF);
+            setEventUrl(EVENT_TYPE_LIGHT, index, config.eventsPush, SHELLY_EVENT_SHORTPUSH1, SHELLY_EVENT_LONGPUSH1,
+                    SHELLY_EVENT_SHORTPUSH2, SHELLY_EVENT_LONGPUSH2);
+
+            // Relay output
+            setEventUrl(EVENT_TYPE_LIGHT, index, config.eventsSwitch, SHELLY_EVENT_OUT_ON, SHELLY_EVENT_OUT_OFF);
+        } else if (profile.hasRelays) {
+            // Standard relays: btn_xxx, out_xxx, short/longpush URLs
+            setEventUrl(EVENT_TYPE_RELAY, index, config.eventsButton, SHELLY_EVENT_BTN_ON, SHELLY_EVENT_BTN_OFF);
+            setEventUrl(EVENT_TYPE_RELAY, index, config.eventsPush, SHELLY_EVENT_SHORTPUSH, SHELLY_EVENT_LONGPUSH);
+            setEventUrl(EVENT_TYPE_RELAY, index, config.eventsSwitch, SHELLY_EVENT_OUT_ON, SHELLY_EVENT_OUT_OFF);
+        } else if (profile.isLight) {
+            // Duo, Bulb
+            setEventUrl(EVENT_TYPE_LIGHT, index, config.eventsSwitch, SHELLY_EVENT_OUT_ON, SHELLY_EVENT_OUT_OFF);
+        }
+    }
+
+    private void setEventUrl(boolean enabled, String... eventTypes) throws ShellyApiException {
+        for (String eventType : eventTypes) {
+            if (profile.containsEventUrl(eventType)) {
+                // Sensors add the type=xx to report_url themself, so we need to ommit here
+                String urlParm = !eventType.equalsIgnoreCase(SHELLY_EVENT_SENSORREPORT) ? "?type=" + eventType : "";
+                String callBackUrl = "http://" + config.localIp + ":" + config.localPort + SHELLY_CALLBACK_URI + "/"
+                        + profile.thingName + "/" + eventType + urlParm;
+                String newUrl = enabled ? callBackUrl : SHELLY_NULL_URL;
+                String test = "\"" + mkEventUrl(eventType) + "\":\"" + newUrl + "\"";
+                if (!enabled && !profile.settingsJson.contains(test)) {
+                    // Don't set URL to null when the current one doesn't point to this OH
+                    // Don't interfere with a 3rd party App
+                    return;
```

Comment:
You forgot to change this
```suggestion
                    continue;
```

Reply:
good catch, thanks